src/main/webapp/app/course/manage/course-management.service.ts

```diff
@@ -136,7 +140,19 @@ export class CourseManagementService {
      */
     findAllForDashboard(): Observable<EntityArrayResponseType> {
         this.fetchingCoursesForNotifications = true;
-        return this.http.get<Course[]>(`${this.resourceUrl}/for-dashboard`, { observe: 'response' }).pipe(
+        return this.http.get<CourseForDashboardDTO[]>(`${this.resourceUrl}/for-dashboard`, { observe: 'response' }).pipe(
+            map((res: HttpResponse<CourseForDashboardDTO[]>) => {
+                if (res.body) {
+                    const courses: Course[] = [];
+                    res.body.forEach((courseForDashboardDTO) => {
+                        courses.push(courseForDashboardDTO.course);
+                        this.saveScoresInStorage(courseForDashboardDTO);
+                    });
+                    // Replace the CourseForDashboardDTOs in the response body with the normal courses to enable further processing.
+                    return res.clone({ body: courses });
+                }
+                return res;
+            }),
             map((res: EntityArrayResponseType) => this.processCourseEntityArrayResponseType(res)),
```

Comment:
Shouldn't we store these courses in the course-storage service here, too, like you do it for the single course in ``findOneForDashboard`` call below?

Reply:
Yes, I think it makes more sense that way. I moved the call to `setCourses()` out of the `courses.component` and into a `tap` for the `findAllForDashboard()` method.